aips/aip-36.md

```diff
@@ -12,15 +12,21 @@ created: 06/01/2023
 
 ## Summary
 
-This AIP proposes to add two new native functions `create_unique_address` and `get_txn_hash` in Aptos framework. The `create_unique_address` generates and outputs a unique 256-bit identifier (of type address) for each function call. The `get_txn_hash` function outputs the hash of the current transaction. The `create_unique_address` function calls can run efficiently and in parallel. In other words, when two transactions run `create_unique_address` method, they can be executed in parallel without any conflicts. Initially, we will use these unique identifiers internally as addresses for newly created Move Objects.
+This AIP proposes to add two new native functions `create_unique_address` and `get_txn_hash` in Aptos framework. The `create_unique_address` generates and outputs a unique 256-bit identifier (of type address) for each function call. The `get_txn_hash` function outputs the hash of the current transaction. The `create_unique_address` function calls can run efficiently and in parallel. In other words, when two transactions run `create_unique_address` method, they can be executed in parallel without any conflicts. Initially, we will use these unique identifiers internally as addresses for newly created Move Objects, specifically in Token V2 minting.
+We also propose to add a wrapper function called `create_uuid` which wraps the unique address generated by the `create_unique_address` in the non-copy-able UUID struct type, guaranteeing that every two instances of UUID value are different.
```

Comment:
is it OK to call this struct `UUID` when it doesn't meet the `UUID` spec (https://datatracker.ietf.org/doc/html/rfc4122)?